deeplinkdispatch-processor/src/main/java/com/airbnb/deeplinkdispatch/Documentor.java

```diff
@@ -122,4 +135,92 @@ private static String formatJavaDoc(String str) {
   File getFile() {
     return file;
   }
+
+  /**
+   * Implement this interface if you want to provide own documentation writer.
+   */
+  interface DocumetationWriter {
+    /**
+     * Compose documentation with help of provided environment, writer and collection of
+     * found deeplink elements.
+     */
+    void write(@Nonnull final ProcessingEnvironment env,
+               @Nonnull final PrintWriter writer,
+               @Nonnull final List<DeepLinkAnnotatedElement> elements);
+  }
+
+  /**
+   * GitHub markdown format.
+   *
+   * @see <a href="https://guides.github.com/features/mastering-markdown/">Mastering Markdown</a>
+   * @see <a href="https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet">Markdown
+   * Cheatsheet</a>
+   * @see <a href="https://github.com/JFormDesigner/markdown-writer-fx">Markdown Writer FX</a>
+   * @see <a href="https://stackedit.io/">Stack Edit</a>
+   */
+  static class MarkdownWriter implements DocumetationWriter {
+
+    @Override
+    public void write(@Nonnull final ProcessingEnvironment env,
+                      @Nonnull final PrintWriter writer,
+                      @Nonnull final List<DeepLinkAnnotatedElement> elements) {
+
+      final Elements utils = env.getElementUtils();
+
+      // header
+      writer.println("| URI | JavaDoc | Simple Name | Method |");
+      writer.println("| --- | ------- | ----------- | ------ |");
+      final String format = "| %1$s | %2$s | %3$s | %4$s |";
+
+      // publish lines
+      for (DeepLinkAnnotatedElement element : elements) {
+        final String uri = element.getUri();
+        final String doc = formatJavaDoc(utils.getDocComment(element.getElement()));
+        final String embeddedComments = null == doc ? "" : doc;
+        final boolean isMethod = element.getAnnotationType().equals(DeepLinkEntry.Type.METHOD);
+        final String methodName = isMethod ? element.getMethod() : "";
+        final Name simpleName = element.getAnnotatedElement().getSimpleName();
+
+        final String line = String.format(Locale.US, format,
+            uri, embeddedComments, simpleName, methodName);
+
+        writer.println(line);
+      }
+
+      writer.flush();
+    }
+  }
+
+  /**
+   * Old documentation format.
+   */
+  static class GenericWriter implements DocumetationWriter {
+    @Override
+    public void write(@Nonnull final ProcessingEnvironment env,
+                      @Nonnull final PrintWriter writer,
+                      @Nonnull final List<DeepLinkAnnotatedElement> elements) {
```

Comment:
Please omit `@Nonnull` annotations since they are unnecessary

Reply:
need here a little details: why its unnecessary?! I always think it helps IDE to detect wrong parameters. `NonNull, Nullable, StringRes, etc.`